odl/discr/diff_ops.py

```diff
@@ -92,12 +121,20 @@ def __init__(self, space, axis=0, method='forward', pad_mode=None,
         # Method is affine if nonzero padding is given.
         linear = not (pad_mode == 'constant' and pad_const != 0)
         super().__init__(domain=space, range=space, linear=linear)
-        self.axis = axis
+        self.axis = int(axis)
         self.dx = space.cell_sides[axis]
-        self.method = method
-        self.pad_mode = pad_mode
-        self.pad_const = pad_const
-        self.edge_order = edge_order
+
+        self.method, method_in = str(method).lower(), method
+        if method not in _SUPPORTED_DIFF_METHODS:
+            raise ValueError('`method` {} not understood'
+                             ''.format(method_in))
+
+        self.pad_mode, pad_mode_in = str(pad_mode).lower(), pad_mode
+        if pad_mode not in _SUPPORTED_PADDING_METHODS:
```

Comment:
Yes, but that's only for really advanced users. I don't want to explicitly disallow it, nor do I want to encourage people to use it. So this is the best I can come up with.